packages/shared/lib/services/connection.service.ts

```diff
@@ -55,6 +56,13 @@ const logger = getLogger('Connection');
 
 type KeyValuePairs = Record<string, string | boolean>;
 
+interface IntegrationValuesWithAccountAndEnvironment {
+    connection_id: string;
+    provider_config_key: string;
+    account: Account;
+    environment: Environment;
+}
+
```

Comment:
not sure declaring an interface is worth it here. Duplicating the struct definition twice is fine imho. You might even even been able to please the typescript compiler with something like this below:
```
public async getOldConnections({
        days,
        limit
    }: {
        days: number;
        limit: number;
    }): Promise<{ connection_id: string; provider_config_key: string; account: Account; environment: Environment }[]> {

        type T = Awaited<ReturnType<ConnectionService['getOldConnections']>>;
        const result = await db
            .knex<StoredConnection>(`_nango_connections`)
...
            .select<T>(
...
```